artiq/coredevice/phaser.py

```diff
@@ -886,6 +958,30 @@ def trf_read(self, addr, cnt_mux_sel=0) -> TInt32:
         return self.trf_write(0x00000008 | (cnt_mux_sel << 27),
                               readback=True)
 
+    @kernel
+    def cal_vco(self):
+        """Start VCO calibration."""
+        self.trf_write(self.trf_mmap[1] | (1 << 31))
+
+    @kernel
+    def en_out(self, rf=1, lo=0):
+        """Enable the rf/lo outputs.
+
+        It is recommended to disable TRF outputs during VCO calibration.
+
+        :param rf: 1 to enable RF output, 0 to disable
+        :param lo: 1 to enable LO output, 0 to disable"""
+        data = self.trf_read(0xc)
```

Comment:
Why not take the value in `trf_mmap` as above?

Reply:
This register contains the `en_frac` bit. Some users may need to reconfigure the VCO frequency in their experiments This may include switching between integer and fractional mode. It seems imprudent to assume this bit is not changed by any user.

Reply:
Ok. The we should make that a rule to use current state as much as possible.